cmake-tool/helpers/rootserver.cmake

```diff
@@ -101,37 +97,28 @@ function(DeclareRootserver rootservername)
             else()
                 set(march rv64imafdc)
             endif()
-            if(UseRiscVBBL)
-                # Package up our final elf image into the Berkeley boot loader.
+            if(UseRiscVOpenSBI)
+                # Package up our final elf image into OpenSBI.
                 # The host string is extracted from the cross compiler setting
                 # minus the trailing '-'
                 if("${CROSS_COMPILER_PREFIX}" STREQUAL "")
                     message(FATAL_ERROR "CROSS_COMPILER_PREFIX not set.")
                 endif()
 
-                string(
-                    REGEX
-                    REPLACE
-                        "^(.*)-$"
-                        "\\1"
-                        host
-                        "${CROSS_COMPILER_PREFIX}"
-                )
-                get_filename_component(host ${host} NAME)
                 file(GLOB_RECURSE deps)
                 add_custom_command(
-                    OUTPUT "${CMAKE_BINARY_DIR}/bbl/bbl"
-                    COMMAND mkdir -p ${CMAKE_BINARY_DIR}/bbl
+                    OUTPUT "${OPENSBI_PATH}/build/platform/sifive/fu540/firmware/fw_payload.elf"
+                    COMMAND mkdir -p ${CMAKE_BINARY_DIR}/opensbi
                     COMMAND
-                        cd ${CMAKE_BINARY_DIR}/bbl && ${BBL_PATH}/configure
-                        --quiet
-                        --host=${host}
-                        --with-arch=${march}
-                        --with-payload=${elf_target_file}
-                            && make -s clean && make -s > /dev/null
+                        cd ${OPENSBI_PATH} && ${CMAKE_OBJCOPY} -O binary ${elf_target_file}
+                        "payload" && make -s clean && CROSS_COMPILE=${CROSS_COMPILER_PREFIX} make
+                        PLATFORM=sifive/fu540 FW_PAYLOAD_PATH=payload
```

Comment:
It seems that this command is for Hifive Unleashed board only. Besides Hifive, we also need to support Ariane, qemu/spike and rocketchip(mapped to ZC706 and ZCU102).

Reply:
Hmm... Is the platform specified somewhere in the build scripts?

OpenSBI supports the Unleashed, Ariane and QEMU. So we could just change the platform. I don't see support for the rocket chip though.

Reply:
Variable ${PLATFORM} contains the name of the targeting platform. Regarding rocket chip, it merely needs HTIF from the bootloader. It seems that OpenSBI has some HTIF support, but I'm not sure how to use it.

Possible values of ${PLATFORM}: ariane, hifive, polarfire, spike, rocketchip.

Polarfire uses U540. I assume it can just use the command for the Unleashed.
Spike is QEMU simulation, it has to support both 64bit and 32bit. We use the "spike" machine for 64bit, but the "virt" machine for 32bit. I couldn't get the 32bit "spike" machine working. Do you have some insights?
```bash
# 32bit simulation, only works for virt machine
qemu-system-riscv32 -machine virt -cpu rv32 -nographic -serial mon:stdio -m size=2000M -bios none -kernel images/sel4test-driver-image-riscv-spike
```

Reply:
I just pushed an update that should handle the different platforms.

I don't have any insights sorry, the Spike machine is rarely used so using the virt machine is probably a better choice anyway.